pkg/logql/syntax/ast.go

```diff
@@ -1825,12 +1832,17 @@ func MergeBinOp(op string, left, right *promql.Sample, filter, isVectorCompariso
 		return res, nil
 	}
 
-	if filter {
-		// if a filter-enabled vector-wise comparison has returned non-nil,
-		// ensure we return the left hand side's value (2) instead of the
-		// comparison operator's result (1: the truthy answer)
+	if notReturnBool {
```

Comment:
The `bool` flag doesn't have anything to do with whether we should perform `swap`. For an example, try `1 < bool $METRIC` in prometheus -- it'll return a vector of values with `0` or `1`.

Reply:
I think it's been there because, swapping has no impact when return `bool` (vector of 0 or 1)

Means, both 
`1 < bool $METRIC` and `$METRIC > bool 1` will return exact same vector of values (0 or 1) without needed to swap. In other words, whether to return `left` or `right` sample  is irrelevant in case of returning bool.